src/Language/PureScript/Environment.hs

```diff
@@ -54,9 +54,10 @@ data TypeClassData = TypeClassData
   { typeClassArguments :: [(Text, Maybe SourceType)]
   -- ^ A list of type argument names, and their kinds, where kind annotations
   -- were provided.
-  , typeClassMembers :: [(Ident, SourceType)]
-  -- ^ A list of type class members and their types. Type arguments listed above
-  -- are considered bound in these types.
+  , typeClassMembers :: [(Ident, SourceType, Maybe [[Text]])]
```

Comment:
Why is this a `Maybe`? Does `Nothing` mean something different from `Just [[]]`?

Reply:
Yes. `Nothing` indicates all type class head variables are referenced in the member so it's not ambiguous. `Just` indicates we need to use VTAs. I'll refactor this to use a data constructor to note the difference.

Reply:
Wouldn't `[[]]` also indicate an unambiguous situation, though?

IIUC, the outer list is disjunctive and the inner list is conjunctive—a use of a member needs to choose *some* set from the outer list, and provide *all* of its variables. If that's right, then `[[]]` means a use of a member needs to choose `[]` and provide all of those variables—no variables, so it's unambiguous.

Reply:
I believe the outer list is always non-empty, even if the type doesn't indicate that. See https://github.com/purescript/purescript/pull/4513/files#diff-face7661808c8804ee47e10a0fd05d5ffc7ddebbef1acccffb5463633e08ae7fL200 for context. `coveringSets` is converted to a `List`, and then mapped over to get `leftOvers`, which is then used to determine whether VTA-requiring args are present via `any null leftOvers` rather than `null leftOvers`. 

I switched it to from a `[[Text]]` to a `Maybe [[Text]]` when my check here (https://github.com/purescript/purescript/pull/4513/files#diff-882f8f8c804b84dfa001aea5a0b9a9d979eb1e99ca4b7d844f790a5bca328e43R443) wasn't using `Nothing`/`Just` to indicate this. If the list was empty, I would use `Nothing` there. I found that using that approach didn't trigger the VTA-required message when it should have.